core/market/src/protocol/discovery.rs

```diff
@@ -235,14 +242,17 @@ impl Discovery {
             };
         }
 
-        // We don't lazy bind broadcasts handlers anymore on first Demand creation.
-        // But we still have option to do this easily in the future.
-        self.bind_gsb_broadcast().await.map_or_else(
-            |e| {
-                log::warn!("Failed to subscribe to broadcasts. Error: {:?}.", e,);
-            },
-            |_| (),
-        );
+        // Only bind broadcasts when re-broadcasts are enabled
+        if self.re_broadcast_enabled() {
+            // We don't lazy bind broadcasts handlers anymore on first Demand creation.
```

Comment:
If re_broadcasts are disabled, we should enable lazy broadcasts

Reply:
this is done in the `core/market/src/matcher.rs`.